File: docs/js/Core2/Medical-Department-1a/MedicalDepartment.js
class MedicalDepartment extends oBJECT {
  constructor({ id, name, nmrOfAvailDoctors}) {
    super( id, name);
    this.nmrOfAvailDoctors = nmrOfAvailDoctors;
    this.plannedExaminations = [];  // a queue
  }
  isDoctorAvailable() {return this.nmrOfAvailDoctors > 0;}
  allocateDoctor() {this.nmrOfAvailDoctors--;}
  releaseDoctor() {this.nmrOfAvailDoctors++;}
}
MedicalDepartment.labels = {"nmrOfAvailDoctors":"avDocs"};

File: Java/Core2/src/main/java/de/oes/core2/medicaldepartament_1b/MedicalDepartment.java
package de.oes.core2.medicaldepartament_1b;

import java.util.ArrayList;

import java.util.List;

import de.oes.core2.foundations.oBJECT;
import de.oes.core2.sim.Simulator;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class MedicalDepartment extends oBJECT{

	private Integer nmrOfAvailDoctors;
	private List<Examination> plannedExaminations;
	
	public MedicalDepartment(Integer id, String name, Simulator sim, Integer nmrOfAvailDoctors) {
		super(id, name, sim);
		this.nmrOfAvailDoctors = nmrOfAvailDoctors;
		this.plannedExaminations = new ArrayList<Examination>();
	}
	
	public boolean isDoctorAvailable() {
		return this.nmrOfAvailDoctors > 0;
	}
	
	public void allocateDoctor() {
		this.nmrOfAvailDoctors--;
	}
	
	public void releaseDoctor() {
		this.nmrOfAvailDoctors++;
	}
}
